### PR TITLE
Fix  listed phpstan errors

### LINKED ISSUE #248: Fix `PHPStan` on level 1
**Scope:**

Currently, running 

```sh
php ../composer.phar run phpstan-analyse
```

returns a couple of errors.

Most of them are related to model changes

```
 ------ ----------------------------------------------------------------------------
  Line   Http/Controllers/Reports.php
 ------ ----------------------------------------------------------------------------
  552    Relation 'meter' is not found in App\Models\Transaction\Transaction model.
  552    Relation 'meter' is not found in App\Models\Transaction\Transaction model.
 ------ ----------------------------------------------------------------------------
```

or mis-use of Laravel objects

```
 ------ --------------------------------------------------------------------------------
  Line   Services/AgentBalanceHistoryService.php
 ------ --------------------------------------------------------------------------------
  58     Called 'last' on Laravel collection, but could have been retrieved as a query.
  72     Called 'avg' on Laravel collection, but could have been retrieved as a query.
  79     Called 'avg' on Laravel collection, but could have been retrieved as a query.
 ------ --------------------------------------------------------------------------------
```

and should be fixed.

**Definition of done:**

The Github Action step `composer run phpstan-analyze` is enabled (i.e. `continue-on-error: true` is removed) and the check passes.

### PR DESCRIPTION
**Pull request by alchalade**
This PR addresses the listed errors of `composer phpstan-analyse`
closes #248

### Pull Request checklist

Please confirm you have completed any of the necessary steps below.

- [x] Meaningful Pull Request title and description
- ~~[ ] Changes tested as described above~~
- ~~[ ] Added appropriate documentation for the change.~~
- ~~[ ] Created GitHub issues for any relevant followup/future enhancements if appropriate.~~

### CODE REVIEW

File: Website/htdocs/mpmanager/app/Services/AgentTransactionService.php [hunk 1]
```diff
@@ -51,6 +51,7 @@ static function ($q) use ($agentId) {
 
     public function getById(int $agentId, ?int $customerId = null): AgentTransaction
     {
+        /* @phpstan-ignore-next-line */
```
Comment (dmohns) [inline Website/htdocs/mpmanager/app/Services/AgentTransactionService.php:54]: Could you please change the ignore here to be more specific?

```suggestion
        // @phpstan-ignore larastan.noUnnecessaryCollectionCall
```

Applies to similar changes in this PR as well

File: Website/htdocs/mpmanager/app/Sms/Bongo.php [hunk 2]
```diff
@@ -77,15 +77,16 @@ public function sendGetSms(string $number, string $body)
...
      * @param string $body
      * @param string $callback
-     *
+     * @param SmsAndroidSetting $smsAndroidSetting
      * @return mixed|StreamInterface
      *
      * @throws \Exception
      */
-    public function sendSms(string $number, string $body, $callback)
+    public function sendSms(string $number, string $body, $callback, \App\Models\SmsAndroidSetting $smsAndroidSetting)
```
Comment (dmohns) [inline Website/htdocs/mpmanager/app/Sms/Bongo.php:89]: Could you import `SmsAndroidSetting` here instead of giving the absolute path?

**Review by dmohns — CHANGES_REQUESTED** [review]
Few comments in the code.

Also, what is still missing is, to enable the CI pipeline that checks PHPStan compliance for future PR's/

File: Website/htdocs/mpmanager/app/Sms/Bongo.php (hunk 2 above)
Comment (alchalade) [reply Website/htdocs/mpmanager/app/Sms/Bongo.php:89]: sure thing

**Pull request closed without merging**